backend/core/src/secondary-adapters/postgres/channel/channel-repository-adapter.ts

```diff
@@ -0,0 +1,73 @@
+import { PrismaClient } from '@prisma/client';
+import { inject, injectable } from 'inversify';
+import { LiveStartResponseDto, StreamingKeyResponseDto } from '~/shared/types/types';
+import { StreamingStatus } from '~/shared/enums/enums';
+import { ChannelRepository } from '~/core/channel/port/channel-repository';
+import { generateUuid } from '~/shared/helpers';
+import { CONTAINER_TYPES } from '~/shared/types/container-type-keys';
+
+@injectable()
+export class ChannelRepositoryAdapter implements ChannelRepository {
+  private prismaClient: PrismaClient;
+
+  constructor(@inject(CONTAINER_TYPES.PrismaClient) prismaClient: PrismaClient) {
+    this.prismaClient = prismaClient;
+  }
+
+  async checkStreamingKey(key: string): Promise<LiveStartResponseDto | null> {
+    const keyRecord = await this.prismaClient.streamingKey.findFirst({
+      where: {
+        key,
+      },
+    });
+    if (!keyRecord) {
+      return null;
+    }
+    const pendingStream = await this.prismaClient.video.findFirst({
+      where: {
+        channelId: keyRecord.channelId,
+        status: StreamingStatus.PENDING,
+      },
+    });
+    if (!pendingStream) {
+      return null;
+    }
+    return {
+      videoId: pendingStream.id,
+      streamingKey: key,
+    };
```

Comment:
The adapters shouldn't do any business logic, they only have getters and setters for infrastructure like databases, message brokers, etc. All business logic should be in services. Please do refactor all methods